### PR TITLE
Travis CI integration

### PR DESCRIPTION
**Pull request by JBBgameich**
This may look a little bit ugly, because all relevant commands are in one line, but to be honest I have no plan of travis and this works fine so far.

I found a nice upload tool [here](https://github.com/probonopd/uploadtool), but I haven't integrated it yet. So all the builds made are just for testing if it builds, not for release.

btw: Please squash my commit mess, you know testing travis ci ... needs a few until it works.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+sudo: required
+
+language: ruby
```
Comment (LNJ2) [inline .travis.yml]: This doesn't seem correct

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+services:
+  - docker
+
+before_install:
+- docker pull debian:sid-slim
```
Comment (LNJ2) [inline .travis.yml]: space is missing at the beginning of the line

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+script:
+- docker run -v $PWD:/mnt/kaidan debian:sid-slim /bin/bash -c "apt update; apt install software-properties-common dirmngr gnupg ca-certificates git -y; apt-key adv --keyserver hkp://pgp.mit.edu --recv-key "210EB0BCA70439F0"; echo deb https://raw.githubusercontent.com/kaidanim/packages/master sid main >> /etc/apt/sources.list; apt-get update; apt-get install cmake build-essential zlib1g-dev libgloox-dev libqt5quick5 libqt5quickcontrols2-5 libqt5quickwidgets5 libqt5qml5 libqt5gui5 libqt5core5a qtdeclarative5-dev qttools5-dev qt5-default qtquickcontrols2-5-dev kirigami2-dev -y; cd /mnt/kaidan; mkdir build; cd build; cmake .. -DI18N=1; make -j$(nproc)"
```
Comment (LNJ2) [inline .travis.yml]: A space is missing at the beginning of the line.
Please separate every command into a new line (starting with ` - `).

**Comment by LNJ2** [comment]
> So all the builds made are just for testing if it builds, not for release.
That's what travis is for, not for really releasing builds.

File: .travis.yml (hunk 3 above)
Comment (LNJ2) [reply .travis.yml]: Alternatively, you can also create a shell script and just execute that.

Comment (JBBgameich) [reply .travis.yml]: >  Please separate every command into a new line (starting with -).
It's only one

Comment (LNJ2) [reply .travis.yml]: Then maybe something like `bash -c $(cat util/travis.sh)`?

Comment (LNJ2) [reply .travis.yml]: Or just `bash -c ./util/travis.sh`

**Comment by JBBgameich** [comment]
https://travis-ci.org/JBBgameich/Kaidan/builds/304347403

**Comment by LNJ2** [comment]
Could you (optionally) move the dependency installation into before-install, instead of script? Then travis-ci would hide it in the log first / you have to unfold it first.

**Comment by JBBgameich** [comment]
Docker seems to reset the container each time before a new command is run inside. You can probably change that, but I don't know how.

**Pull request merged**